sklearn/base.py

```diff
@@ -109,12 +115,8 @@ def clone(estimator, safe=True):
                     and param1.shape == param2.shape
                 )
         else:
-            new_obj_val = new_object_params[name]
-            params_set_val = params_set[name]
-            # The following construct is required to check equality on special
-            # singletons such as np.nan that are not equal to them-selves:
-            equality_test = (new_obj_val == params_set_val or
-                             new_obj_val is params_set_val)
+            # fall back on standard equality
+            equality_test = param1 == param2
```

Comment:
I get it: the "is" test has already been perform above, right?

I am slow.

Maybe a test case with a nan would be a good thing.